docs/zh/latest/discovery/dns.md

```diff
@@ -0,0 +1,131 @@
+---
+title: DNS
+---
+
+<!--
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+* [通过 DNS 服务发现](#通过 DNS 服务发现)
+    * [SRV 记录](#SRV 记录)
+
+## 通过 DNS 服务发现
+
+某些服务发现系统如 Consul，支持通过 DNS 提供系统信息。我们可以使用这种方法直接实现服务发现。
+
+首先我们需要配置 DNS 服务器的地址：
+
+```yaml
+# 添加到 config.yaml
+discovery:
+   dns:
+     servers:
+       - "127.0.0.1:8600"          # 使用 DNS 服务器的真实地址
+```
+
+与配置 Upstream `nodes` 项不同的是，DNS 服务发现将返回所有的记录。例如按照以下的 upstream 配置：
+
+```json
+{
+    "id": 1,
+    "discovery_type": "dns",
+    "service_name": "test.consul.service",
+    "type": "roundrobin"
+}
+```
+
+之后 `test.consul.service` 将被解析为 `1.1.1.1` 和 `1.1.1.2`，这个结果等同于：
+
+```json
+{
+    "id": 1,
+    "type": "roundrobin",
+    "nodes": [
+        {"host": "1.1.1.1", "weight": 1},
+        {"host": "1.1.1.2", "weight": 1}
+    ]
+}
+```
+
+注意所有来自 `test.consul.service` 的 IP 都有相同的权重。
+
+如果一个服务同时有 A 和 AAAA 记录，那么 A 记录的优先级更高。
```

Comment:
It seems we should also specify the precedence between `A` and `SRV`.

Reply:
I would appreciate it if you make supplement on the document in English before I update it?

https://github.com/apache/apisix/blame/master/docs/en/latest/discovery/dns.md#L69

Reply:
I just make a PR for you:
https://github.com/apache/apisix/pull/5033/files